### PR TITLE
Implementing dto`s classes

### PR DESCRIPTION
**Pull request by DPotapov93**

### CODE REVIEW

**Review by DevIgork — APPROVED** [review]
I think you did a good job.

File: pom.xml [hunk 1]
```diff
@@ -9,13 +9,20 @@
         <relativePath/> <!-- lookup parent from repository -->
     </parent>
     <groupId>mate.academy</groupId>
-    <artifactId>online-book-store01</artifactId>
+    <artifactId>book-shop</artifactId>
```
Comment (bhdnchui) [inline pom.xml:12]: That task requires such significant changes.
Please, make changes that are required by task only.

File: src/main/java/mate/academy/onlinebookstore01/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@RequestMapping(value = "/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/api/books/{id}")
```
Comment (bhdnchui) [inline src/main/java/mate/academy/onlinebookstore01/controller/BookController.java:26]: You already defined `/api/books` part in `@RequestMapping` annotation.
Did you test your solution?
```suggestion
    @GetMapping("/{id}")
```

File: src/main/java/mate/academy/onlinebookstore01/service/BookServiceImpl.java [hunk 3]
```diff
@@ -10,18 +14,31 @@
...
-    public BookServiceImpl(BookRepository bookRepository) {
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
         this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.toModel(createBookRequestDto);
+        return bookMapper.toDto(book);
```
Comment (bhdnchui) [inline src/main/java/mate/academy/onlinebookstore01/service/BookServiceImpl.java:28]: Where is `bookRepository.save()` method call?

File: src/main/java/mate/academy/onlinebookstore01/service/BookServiceImpl.java [hunk 4]
```diff
@@ -10,18 +14,31 @@
...
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find book by id: " + id));
+        return bookMapper.toDto(book);
```
Comment (bhdnchui) [inline src/main/java/mate/academy/onlinebookstore01/service/BookServiceImpl.java:42]: Rewrite it and avoid redundant temporary `Book book` variable.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Please, pay attention to my comments.

File: src/main/java/mate/academy/onlinebookstore01/repository/BookRepositoryImpl.java [hunk 5]
```diff
@@ -49,4 +50,12 @@ public List<Book> findAll() {
             throw new RuntimeException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/repository/BookRepositoryImpl.java:58]:
```suggestion
            return Optional.ofNullable(session.find(Book.class, id));
```

**Review by d1sam — APPROVED** [review]
Well done, but be careful with unnecessary variables.

**Pull request merged**